File: sorting_algorithms/bubble_sort.rb
# Completed, not refactored

# In bubble sort, adjacent pairs are compared
# smaller numbers move down, bigger ones move up 

def bubble_sort(num_array_to_sort)

    # do this for as many times as there are elements in the array
    for j in 1...num_array_to_sort.length   

        num_array_to_sort.each_with_index do |num, i|
            if i+1 == num_array_to_sort.length
                break
            elsif num_array_to_sort[i] > num_array_to_sort[i+1]
                temp = num_array_to_sort[i]
                num_array_to_sort[i] = num_array_to_sort[i+1]
                num_array_to_sort[i+1] = temp
            end
        end
        j +=1

    end
    
    num_array_to_sort  # returns sorted array

end

a = (0..15).to_a.shuffle!
p a.inspect
p bubble_sort(a)

File: sorting_algorithms/bubble_sort.js
// Completed, not refactored

function bubble_sort(array_to_sort) {
    var num_loops = array_to_sort.length;
    var sorted_array = "placeholder";
    for(var i = 1; i < num_loops; i++){
        for(var j = 1; j < num_loops; j++) {
            if(array_to_sort[(j-1)] > array_to_sort[j]) {
                var switched = swap(array_to_sort[(j-1)], array_to_sort[j]);
                array_to_sort[(j-1)]=switched[0];
                array_to_sort[(j)]=switched[1];
            }
        }
    }
    return array_to_sort;
}

function swap(a, b) {
    var temp = a;
    a = b;
    b = temp;
    return [a, b];
}

a = [2, 5, 6, 8, 7, 9, 2, 4, 3, 0, 6];
bubble_sort(a);